pipeline/backend/kubernetes/pod.go

```diff
@@ -93,6 +93,13 @@ func podMeta(step *types.Step, config *config, podName string) (metav1.ObjectMet
 		meta.Labels[ServiceLabel] = step.Name
 	}
 
+	if val, ok := step.Environment["CI_REPO_NAME"]; ok {
+		meta.Labels["repository"] = val
+	}
+	if val, ok := step.Environment["CI_PIPELINE_NUMBER"]; ok {
+		meta.Labels["pipeline_number"] = val
+	}
```

Comment:
I propose to make separate map `affinityLabels` in `mlPod()` and supply it to `podMeta` and `podSpec` then.
Also, isn't there some pipeline ID besides repo-name/pipeline-number pair?

Reply:
I don't get it, is better to do it in the podMeta? is related to metadata as well. Then podSpec receives the labels and mounts the affinity block.

Reply:
Yeah, but podSpec receives all labels then... not just required affinity ones...
And also we can wrap labels creation in `if` and do not label Pod at all:
```
affinityLabels = createAffinityLabels(step)
podMeta(..., affinityLabels)
podSpec(..., affinityLabels)
```
```
func createAffinityLabels(step) map {
  affinityLabels : map
  if !config.StorageRwx { # or whatever condition, config maybe
    # fill affinity map from environment 
  }
 return affinityLabels
}
```
Now we always label Pod. Don't know, if it's bad or not, though.